core/opendaq/security/include/opendaq/permission_config_builder.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <coretypes/baseobject.h>
+#include <coretypes/common.h>
+#include <opendaq/permission_config.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*!
+ * @ingroup opendaq_security
+ * @addtogroup opendaq_security_permission_config_builder PermissionConfigBuilder
+ * @{
+ */
+
+/*!
+ * @brief A class which is responsible for assigning permissions to a property object.
+ * Permisison builder can specified allowed permissions for each group. It can also inherit
+ * or overwrite premissions from parent objects.
+ */
+DECLARE_OPENDAQ_INTERFACE(IPermissionConfigBuilder, IBaseObject)
+{
+    // [returnSelf]
+    /*!
+     * @brief Configure component to inherit or ignore permissions from parent object.
+     * @param inherit Flag if components should inherit permissions from it's parent object.
+     */
+    virtual ErrCode INTERFACE_FUNC inherit(Bool inherit) = 0;
+
+    // [returnSelf]
+    /*!
+     * @brief Set prmissions for given group.
+     * @param groupId The id of a group to set permissions for.
+     * @param permissionFlags Bit mask of permissions to allow or deny.
```

Comment:
I'm maybe missing a comment that links the `permissions` enum to the bit mask. Basically some guideline on how to construct the bitmask.

Reply:
Updated